idea/src/org/jetbrains/kotlin/idea/quickfix/AssignToPropertyFix.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.quickfix
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import org.jetbrains.kotlin.descriptors.CallableDescriptor
+import org.jetbrains.kotlin.diagnostics.Diagnostic
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+import org.jetbrains.kotlin.resolve.BindingContext
+
+class AssignToPropertyFix(element: KtNameReferenceExpression) : KotlinQuickFixAction<KtNameReferenceExpression>(element) {
+
+    override fun getText() = "Assign to property"
+
+    override fun getFamilyName() = text
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val element = element ?: return
+        val psiFactory = KtPsiFactory(element)
+        element.replace(psiFactory.createExpressionByPattern("this.$0", element))
+    }
+
+    companion object : KotlinSingleIntentionActionFactory() {
+        override fun createAction(diagnostic: Diagnostic): KotlinQuickFixAction<KtNameReferenceExpression>? {
+            val expression = diagnostic.psiElement as? KtNameReferenceExpression ?: return null
+
+            val containingClass = expression.containingClass() ?: return null
+            val right = (expression.parent as? KtBinaryExpression)?.right ?: return null
+            val type = expression.analyze().getType(right) ?: return null
+            val name = expression.getReferencedNameAsName()
+
+            val hasAssignableProperty = containingClass.getProperties().any {
+                it.nameAsName == name &&
+                it.isVar &&
```

Comment:
I think there is a reason to improve this case for secondary constructors. Inside them, `var` is not required if property does not have its own initializer and is not declared in primary constructor. Code like `foo = foo` is high probable in this case.

Reply:
Fixed

Reply:
Also added a fix for property defined in primary constructor.
```kotlin
class Test(var foo: Int) {
    fun test(foo: Int) {
        <caret>foo = foo
    }
}
```